sdk/trace/src/test/java/io/opentelemetry/sdk/trace/internal/AlwaysRecordSamplerTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package io.opentelemetry.sdk.trace.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.Collections;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/** Unit tests for {@link AlwaysRecordSampler}. */
+class AlwaysRecordSamplerTest {
+
+  // Mocks
+  private Sampler mockSampler;
+
+  private AlwaysRecordSampler sampler;
+
+  @BeforeEach
+  void setUpSamplers() {
+    mockSampler = mock(Sampler.class);
+    sampler = AlwaysRecordSampler.create(mockSampler);
+  }
+
+  @Test
+  void testGetDescription() {
+    when(mockSampler.getDescription()).thenReturn("mockDescription");
+    assertThat(sampler.getDescription()).isEqualTo("AlwaysRecordSampler{mockDescription}");
+  }
+
+  @Test
+  void testRecordAndSampleSamplingDecision() {
```

Comment:
Can use a `org.junit.jupiter.params.ParameterizedTest` test to improve readability.